### DISCUSSION TITLE
What does "Error: Expected at least one matching middleware" mean?

### DISCUSSION THREAD

**Discussion by Ethelt** [vercel/community#105] [Help] [answered]
Hi, can anyone tell me what does "Error: Expected at least one matching middleware" in my Next,js middleware function mean? I couldn't find any info on it on the internet, or in Vercel and Next.js code. 
It happens every time I try to access a page with English locale (en-us, en-gb, etc.), but not in other languages (pl-us works fine), unless I set the English locale as a default one. Everything works fine on my machine, but it breaks when trying to deploy it on Vercel. It happens regardless of middleware function content.

**Comment by leerob** [comment]
Could you share a link to your repository so we can take a look? Thank you!

**Reply by Ethelt** [reply]
Thanks for reply, here's a link to a barebones version of the project, where I can still reproduce the error: https://github.com/Ethelt/middleware-test/
If you want to take a look at my deployment, it's here: https://middleware-test-hgqn9vrja-gd-kf820nsm-origin-getdressedm.vercel.app
You can reproduce the error by going to /en-us.
"Error: Expected at least one matching middleware" is the only logging I get, though interestingly it doesn't show up in Errors tab of realtime logs.

**Reply by leerob** [reply]
Thank you! We have an open PR to fix this, should be resolved very soon.

**Reply by mismosmi** [reply]
I see the same issue. Would you let me know if this is resolved?

Thanks!

**Reply by rafael-franca** [reply]
Me too. When I added a middleware in root.

**Reply by khrome83** [reply]
I am seeing this also with the `ENABLE_FILE_SYSTEM_API` as well. Can you link to the PR @leerob so we can track when this is resolved?

**Reply by leerob** [reply]
Why are you using `ENABLE_FILE_SYSTEM_API`? Could you share more?

**Comment by Pieparker** [comment, marked as answer]
This conversation appears to have gone quiet. To keep discussions on topic and to prevent the resurrection of old topics, this discussion has been closed. If issues persist or a future reader stumbles upon this discussion and is exhibiting the same problem, please create a new discussion.
